spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,13 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
         after(:each) do
           Puppet.settings[:environment_timeout] = 0
           Puppet.lookup(:environments).clear_all
```

Comment:
Also if we should be cleaning up `environments` global state for each test, then we probably should put that in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/test/test_helper.rb#L151